src/net.cpp

```diff
@@ -841,36 +872,126 @@ const uint256& CNetMessage::GetMessageHash() const
 
 
 
+size_t CConnman::SocketReceiveData(CNode* pnode)
+{
+    // typical socket buffer is 8K-64K
+    char pchBuf[0x10000];
+    int nBytes = recv(pnode->hSocket, pchBuf, sizeof(pchBuf), MSG_DONTWAIT);
+    if (nBytes > 0)
+    {
+        bool notify = false;
+        if (!pnode->ReceiveMsgBytes(pchBuf, nBytes, notify))
+            pnode->Disconnect();
 
+        auto it(pnode->vRecvMsg.begin());
+        size_t nSizeAdded = 0;
+        {
+            LOCK(pnode->cs_vRecv);
+            int64_t nTimeMicros = GetTimeMicros();
+            pnode->nLastRecv = nTimeMicros / 1000000;
+            pnode->nRecvBytes += nBytes;
+            if (notify) {
+                for (; it != pnode->vRecvMsg.end(); ++it) {
+                    const auto& msg = *it;
+                    if (!msg.complete())
+                        break;
+                    nSizeAdded += msg.vRecv.size() + CMessageHeader::HEADER_SIZE;
+                    mapMsgCmdSize::iterator mapit = pnode->mapRecvBytesPerMsgCmd.find(msg.hdr.pchCommand);
+                    if (mapit == pnode->mapRecvBytesPerMsgCmd.end())
+                        mapit = pnode->mapRecvBytesPerMsgCmd.find(NET_MESSAGE_COMMAND_OTHER);
+                    assert(mapit != pnode->mapRecvBytesPerMsgCmd.end());
+                    mapit->second += msg.vRecv.size() + CMessageHeader::HEADER_SIZE;
+                }
+            }
+        }
+        if (it != pnode->vRecvMsg.begin()) {
+            LOCK(pnode->cs_vProcessMsg);
+            pnode->vProcessMsg.splice(pnode->vProcessMsg.end(), pnode->vRecvMsg, pnode->vRecvMsg.begin(), it);
+            pnode->nProcessQueueSize += nSizeAdded;
+            if (pnode->nProcessQueueSize > nReceiveFloodSize) {
+                LogPrint(BCLog::NET, "Disabling receive due to oversized process queue of size: %lu. peer=%i\n", pnode->nProcessQueueSize, pnode->GetId());
+                pnode->fPauseRecv = true;
+                event_del(pnode->m_read_event);
+            }
 
+        }
+        WakeMessageHandler();
```

Comment:
Should this be moved into the above `if` block? (It used to be inside the `if (notify)` statement)

Reply:
Yes, thank you.